visidata/canvas.py

```diff
@@ -676,10 +681,10 @@ def render_sync(self):
 Canvas.addCommand('t', 'stoggle-cursor', 'source.toggle(list(rowsWithin(plotterCursorBox)))')
 Canvas.addCommand('u', 'unselect-cursor', 'source.unselect(list(rowsWithin(plotterCursorBox)))')
 Canvas.addCommand(ENTER, 'dive-cursor', 'vs=copy(source); vs.rows=list(rowsWithin(plotterCursorBox)); vd.push(vs)')
-Canvas.addCommand('d', 'delete-cursor', 'toDelete=list(rowsWithin(plotterCursorBox)); source.copyRows(toDelete); source.deleteBy(lambda r,toDelete=toDelete: r in toDelete); reload()')
+Canvas.addCommand('d', 'delete-cursor', 'toDelete=list(rowsWithin(plotterCursorBox)); deleteCursor(toDelete); reload()')
```

Comment:
- The reload() needs to be inside deleteCursor(), so it runs properly after the delete has completed (otherwise it may reload the Canvas from a Sheet with a delete-in-progress.
- toDelete doesn't have to be a separate variable anymore.
- the list() should be done inside deleteCursor, because it will need to become a dict in order to achieve O(N*logM).  better to pass the iterator from rowsWithin() directly.

Reply:
Can an async function be called within an async function?

Reply:
"toDelete doesn't have to be a separate variable anymore."

Should there then be a Canvas().deleteCursor() and Canvas().deleteVisible()?

Reply:
I guess it should be `Canvas.deleteSourceRows()` and then you pass the rowsWithin() to it.  Or maybe it should be `Canvas.deleteBox()` and do the `rowsWithin` itself?

Reply:
> Can an async function be called from within an async function?

Yes, of course :)